tmd/Neuron/conversion.py

```diff
@@ -0,0 +1,127 @@
+"""Functions for converting morphio to tmd Neuron
+"""
+from collections import namedtuple
+import numpy as np
+from tmd.Soma.Soma import Soma
+from tmd.Tree import Tree
+from tmd.utils import tree_type as TREE_TYPES
+
+
+SectionData = namedtuple('SectionData', ['points', 'diameters', 'section_type', 'parents'])
+
+
+def convert_morphio_soma(morphio_soma):
+    """Converts a morphio's morphology
+    Args:
+        morphio_soma (Union[morphio.Soma])
+    Returns:
+        tmd_soma (Soma)
+    """
+    points = morphio_soma.points
+    return Soma(
+        x=points[:, 0],
+        y=points[:, 1],
+        z=points[:, 2],
+        d=morphio_soma.diameters
+    )
+
+
+def _section_to_data(section, tree_length, start, parent):
+    """Extract data from morphio section
+
+    Args:
+        section (morphio.Section): A morphio section object
+        tree_length (int): Number of nodes in the tree so far
+
+    Returns:
+        tuple:
+            - n (int): Number of nodes extracted
+            - section_data (SectionData): namedtuple with section data
+    """
+    points = section.points
+
+    # start is either 0 or 1 if the section is root or not respectively
+    # dropping the first node in the latter case
+    n = len(points) - start
+
+    # each node has the previous node as a parent, except for the first
+    # one, which is given from the parent
+    parents = np.arange(tree_length - 1, tree_length + n - 1, dtype=np.int)
+    parents[0] = parent
+
+    return n, SectionData(
+        points[start:],
+        section.diameters[start:],
+        int(section.type),
+        parents)
+
+
+def convert_morphio_trees(morphio_neuron):
+    """Convert morphio morphology's trees to tmd ones
+
+    Args:
+        morphio_neuron (Union[morphio.Morphology, morphio.mut.Morphology]):
+            morphio neuron object
+
+    Yields:
+        tuple:
+            - tree (Tree): The constructed tmd Tree
+            - tree_types (dict): The neuron tree types
+    """
+    total_points = len(morphio_neuron.diameters)
+
+    x = np.empty(total_points, dtype=np.float32)
+    y = np.empty(total_points, dtype=np.float32)
+    z = np.empty(total_points, dtype=np.float32)
+    d = np.empty(total_points, dtype=np.float32)
```

Comment:
After discussion with @wizmer I think it makes sense to use float32. However, the default behavior of numpy seems to be using float64. Can we make this consistent in the tree constructor of tmd?